client/rpc/src/lib.rs

```diff
@@ -0,0 +1,144 @@
+use jsonrpsee::core::async_trait;
+use jsonrpsee::core::RpcResult;
+use jsonrpsee::proc_macros::rpc;
+use jsonrpsee::types::error::ErrorObjectOwned as JsonRpseeError;
+use jsonrpsee::types::error::INTERNAL_ERROR_CODE;
+use jsonrpsee::types::error::INTERNAL_ERROR_MSG;
+use jsonrpsee::types::ErrorObjectOwned;
+
+use shc_common::types::FileMetadata;
+use shc_common::types::FILE_CHUNK_SIZE;
+use sp_runtime::AccountId32;
+use sp_trie::TrieLayout;
+
+use file_manager::traits::FileDataTrie;
+use file_manager::traits::FileStorage;
+
+use log::debug;
+use log::error;
+
+use std::fmt::Debug;
+use std::fs::File;
+use std::io::Read;
+use std::marker::PhantomData;
+use std::path::PathBuf;
+use std::sync::Arc;
+
+use tokio::sync::RwLock;
+
+const LOG_TARGET: &str = "file-storage-rpc";
+
+/// Provides an interface with the desired RPC method.
+/// Used by the `rpc` macro from `jsonrpsee`
+/// to generate the trait that is actually going to be implemented.
+#[rpc(server, namespace = "filestorage")]
+#[async_trait]
+pub trait FileStorageApi {
+    #[method(name = "uploadFile")]
+    async fn upload_file(
+        &self,
+        file_path: String,
+        location: String,
+        owner: AccountId32,
+    ) -> RpcResult<FileMetadata>;
+}
+
+/// Stores the required objects to be used in our RPC method.
+pub struct FileStorageRpc<FL, T> {
+    file_storage: Arc<RwLock<FL>>,
+    _marker: PhantomData<T>,
+}
+
+impl<FL, T> FileStorageRpc<FL, T> {
+    pub fn new(file_storage: Arc<RwLock<FL>>) -> Self {
+        Self {
+            file_storage,
+            _marker: Default::default(),
+        }
+    }
+}
+
+/// Interface generated by the `rpc` macro from our `FileStorageApi` trait.
+#[async_trait]
+impl<FL, T> FileStorageApiServer for FileStorageRpc<FL, T>
+where
+    FL: Send + Sync + FileStorage<T>,
+    T: Send + Sync + TrieLayout + 'static,
+{
+    async fn upload_file(
```

Comment:
I think we either call this 'load_file_in_store' or also add some mechanism to watch for this file.

I think right now the `UserSendsFileTask` processes all events and will just panic if the file is not found in `file_storage`.

Reply:
Makes sense, I'll rename and add a TODO here